kafka/topiccreator.go

```diff
@@ -180,37 +204,37 @@ func (c *TopicCreator) CreateTopics(ctx context.Context, topics ...apmqueue.Topi
 				zap.String("topic", topicName),
 			)
 		}
-		if len(c.topicConfigs) > 0 {
-			alterCfg := make([]kadm.AlterConfig, 0, len(c.topicConfigs))
-			for k, v := range c.topicConfigs {
-				alterCfg = append(alterCfg, kadm.AlterConfig{Name: k, Value: v})
-			}
-			alterResp, err := c.m.adminClient.AlterTopicConfigs(ctx, alterCfg,
-				existingTopics...,
+	}
+	if len(c.topicConfigs) > 0 {
+		alterCfg := make([]kadm.AlterConfig, 0, len(c.topicConfigs))
+		for k, v := range c.topicConfigs {
+			alterCfg = append(alterCfg, kadm.AlterConfig{Name: k, Value: v})
+		}
+		alterResp, err := c.m.adminClient.AlterTopicConfigs(ctx,
+			alterCfg, topicNames...,
```

Comment:
Should we only be altering topic configs for topics that we didn't just create? We'll get a lot of log spam otherwise won't we?

Reply:
We could, I didn't as to decouple the two... But I suppose we can. I can change it if you think it's significantly better. If it's just due to the logging, we could consolidate the info logging to a single line and use a topic slice instead of a single log line per topic altered.

Reply:
Even if it's one log line vs. N, I'd rather not see an Info log every time this function is called if it's effectively a no-op. That aside, it would be good to minimise the number of network calls - that's a smaller issue though, considering how infrequently this should be called.

Reply:
I've updated the code and the test a bit in 8acf0af. Let me know what you think.